Make TIDSID related code work with French games

remicalixte opened the pull request:
- Fix addresses of `FUN_LoadAdrsAdrs` and `FUN_CompAdrsAdrs` for French and other european languages in TIDSID editor
- Add support for French in TIDSID randomizer: add a parameter for the rng call address to adapt to other languages than English


I tested both codes on my French Diamond version

remicalixte committed:
tidsid editor: fix european languages addresses

remicalixte committed:
tidsid randomizer: add french support

DevreeseJorik reviewed (changes requested):
Hi! Thanks for opening this PR. I appreciate the help!

Since the start of this converter I've actually written a script to generate the addresses for all script command functions. I see that the change for the addresses you did for french do indeed match what I dumped. Could you expand this including the pointers for all languages rather than only french? I've put the dumper script along with all versions pre-dumped in a convenient repository:
https://github.com/DevreeseJorik/DP_ScriptCommandDumper

Secondly, since variables take a lot of space and get messy quickly, could you store the entire rng_call value, and bitshift out the bytes in the script section?

Feel free to ask questions or say this is out of scope, in which case I can update it further from here. Thanks!

remicalixte committed:
add japanese rev5 to tidsid editor

remicalixte committed:
add partial support other language in tidsid randomizer

remicalixte commented:
I added japanese rev5 to the tidsid editor thanks to your dumps. I didn't add japanese rev6 because it is missing base_address and I am not sure yet how to get it.

For the randomizer I added the other languages as well and used shifts and masks to convert the rng call address. For this one I only got the French value from searching patterns in the memory editor. Is there a better way to get it? Otherwise I'll do the same for the other languages when I have some time.
